clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -469,19 +469,33 @@ public void resetTimer() {
             this.heartbeatTimer.reset(heartbeatIntervalMs);
         }
 
+        /**
+         * Check if a heartbeat request should be sent on the current time. A heartbeat should be
+         * sent if the heartbeat timer has expired, backoff has expired, and there is no request
+         * in-flight.
+         */
         @Override
         public boolean canSendRequest(final long currentTimeMs) {
             update(currentTimeMs);
             return heartbeatTimer.isExpired() && super.canSendRequest(currentTimeMs);
         }
 
-        public long nextHeartbeatMs(final long currentTimeMs) {
+        public long timeToNextHeartbeatMs(final long currentTimeMs) {
             if (heartbeatTimer.remainingMs() == 0) {
```

Comment:
Sorry if I comment on code outside the PR. Isn't this the same as `heartbeatTimer.isExpired()`? If yes, could we please change this to make the the code more readable?

Reply:
They achieve the same here, and totally agree that `isExpired` is more readable, fixed. (Sensible "since we're here..." to me too btw)